go/client/wallet_api_handler.go

```diff
@@ -512,3 +570,66 @@ func (c *batchOptions) Check() error {
 
 	return nil
 }
+
+type sendPathPaymentOptions struct {
+	Recipient        string `json:"recipient"`
+	Amount           string `json:"amount"`
+	SourceAsset      string `json:"source-asset"`
+	DestinationAsset string `json:"destination-asset"`
+	FromAccountID    string `json:"from-account-id"`
+	Message          string `json:"message"`
+	MemoText         string `json:"memo-text"`
+}
+
+func (c *sendPathPaymentOptions) Check() error {
+	// Note: we don't validate assets in here, since we need the parsed asset values in `sendPathPayment`
+
+	if strings.TrimSpace(c.Recipient) == "" {
+		return ErrRecipientMissing
+	}
+	if strings.TrimSpace(c.Amount) == "" {
+		return ErrAmountMissing
+	}
+
+	namt, err := stellarnet.ParseStellarAmount(c.Amount)
+	if err != nil {
+		return ErrInvalidAmount
+	}
+	if namt < 0 {
+		return ErrInvalidAmount
+	}
+
+	if c.FromAccountID != "" {
+		_, err := strkey.Decode(strkey.VersionByteAccountID, c.FromAccountID)
+		if err != nil {
+			return ErrInvalidAccountID
+		}
+	}
+	if len(c.Message) > libkb.MaxStellarPaymentNoteLength {
+		return ErrMessageTooLong
+	}
+	if len(c.MemoText) > libkb.MaxStellarPaymentPublicNoteLength {
+		return ErrMemoTextTooLong
+	}
+
+	return nil
+}
+
+func parseAsset(s string) (stellar1.Asset, error) {
```

Comment:
This function is nearly identical to `parseAssetFlag` in `cmd_wallet_send_path_payment.go`. Would love to consolidate them if you have any suggestions as to how!

Reply:
you could put it in a new file `wallet_common.go` keeping it lowercase.  Then both `cmd_wallet_send_path_payment.go` and this could access it.